@here/harp-mapview/lib/Watermark.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as THREE from "three";
+
+/**
+ * Creates and handles rendering of watermark logo geometry for [[MapView]].
+ */
+export class Watermark {
+    private m_watermarkMesh: THREE.Mesh;
+    private readonly WATERMARK_RENDERORDER = 100000;
+    private readonly m_watermarkScene = new THREE.Scene();
+
+    /**
+     * Default constructor.
+     *
+     * @param textureUrl URL to watermark image.
+     * @param m_size Watermark size.
+     * @param m_margin Distance from canvas corner to watermark logo.
+     */
+    constructor(textureUrl: string, private m_size = 32, private m_margin = 4) {
+        const geometry = new THREE.PlaneGeometry(this.m_size, this.m_size);
+        const map = new THREE.TextureLoader().load(textureUrl);
+        const material = new THREE.MeshBasicMaterial({
```

Comment:
I think https://threejs.org/docs/index.html#api/en/objects/Sprite and 
https://threejs.org/docs/index.html#api/en/materials/SpriteMaterial would be mare appropriate for this usecase.

Reply:
If you have repo version checked out, search for: terrain_raycast-and-getheight.html

I render a textured sprite there.